### PR TITLE
Job: Replace deprecated wait functions with supported one

### PR DESCRIPTION
**Pull request by tenzen-y**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
I replaced the deprecated wait functions with the supported one.

Deprecated: `PollImmediate` and `Poll`:

https://github.com/kubernetes/kubernetes/blob/dc8b57d8a75764e649d641ac77c1539a574d9aed/staging/src/k8s.io/apimachinery/pkg/util/wait/poll.go#L142-L158

https://github.com/kubernetes/kubernetes/blob/dc8b57d8a75764e649d641ac77c1539a574d9aed/staging/src/k8s.io/apimachinery/pkg/util/wait/poll.go#L51-L67

New: `PollUntilContextTimeout`:

https://github.com/kubernetes/kubernetes/blob/dc8b57d8a75764e649d641ac77c1539a574d9aed/staging/src/k8s.io/apimachinery/pkg/util/wait/poll.go#L36-L49

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by tenzen-y** [comment]
The failure isn't related to this PR: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/121323/pull-kubernetes-e2e-gce/1714630128991997952

/pull-kubernetes-e2e-gce

**Comment by tenzen-y** [comment]
Test pod wasn't scheduled to any nodes: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/121323/pull-kubernetes-e2e-gce/1714649244142407680

/test pull-kubernetes-e2e-gce

File: test/integration/job/job_test.go [hunk 1]
```diff
@@ -2305,7 +2305,7 @@ func TestJobFailedWithInterrupts(t *testing.T) {
 		t.Fatalf("Could not fail a pod: %v", err)
 	}
 	remaining := 9
-	if err := wait.PollImmediate(5*time.Millisecond, wait.ForeverTestTimeout, func() (done bool, err error) {
```
Comment (mimowo) [inline test/integration/job/job_test.go:2308]: Why the change from `err := ` to `err =` ?

File: test/integration/job/job_test.go [hunk 2]
```diff
@@ -2507,7 +2507,7 @@ func TestNodeSelectorUpdate(t *testing.T) {
 	// (2) Check that the pod was created using the expected node selector.
 
 	var pod *v1.Pod
-	if err := wait.PollImmediate(waitInterval, wait.ForeverTestTimeout, func() (bool, error) {
```
Comment (mimowo) [inline test/integration/job/job_test.go:2510]: Why the change from `err := ` to `err =` ?

**Review by mimowo — COMMENTED** [review]
LGTM, questions about the err variable reuse. 
Thanks for taking this.

File: test/integration/job/job_test.go (hunk 1 above)
Comment (tenzen-y) [reply test/integration/job/job_test.go:2308]: `err` is already defined in the above line (L2284). So I think we can avoid re-defining `err` using `:=`. 
This means it would be better to avoid shadow declaration.

WDYT?

File: test/integration/job/job_test.go (hunk 2 above)
Comment (tenzen-y) [reply test/integration/job/job_test.go:2510]: ditto.

File: test/integration/job/job_test.go (hunk 1 above)
Comment (tenzen-y) [reply test/integration/job/job_test.go:2308]: It makes sense. It would be helpful to track the changes.

Comment (tenzen-y) [reply test/integration/job/job_test.go:2308]: Done.

File: test/integration/job/job_test.go (hunk 2 above)
Comment (tenzen-y) [reply test/integration/job/job_test.go:2510]: Done.

**Review by mimowo — COMMENTED** [review]
/lgtm
/assign @alculquicondor

**Review by alculquicondor — COMMENTED** [review]
/approve
